test/test_buffer.cpp

```diff
@@ -152,7 +152,7 @@ void free_buffer(char* m)
 
 char* allocate_buffer(int size)
 {
-	char* mem = (char*)std::malloc(size);
+	char* mem = static_cast<char*>(std::malloc(std::size_t(size)));
```

Comment:
I wonder if there's a reason not to use ``new char[size]`` here instead, and avoid the casts

Reply:
if the ``chained_buffer`` frees with ``free``, this would still have to be ``malloc``

Reply:
OK

Reply:
yes, `std::free` is used with the `free_buffer` above, no change here